src/client_adapter/headers_in_memory.rs

```diff
@@ -32,12 +32,18 @@ impl Blockchain for HeadersInMemory {
         unimplemented!()
     }
 
-    fn best_block_header(&self) -> Option<BlockHeader> { unimplemented!() }
+    fn best_block_header(&self) -> Option<&BlockNumber> {
+        self.headers.keys().max()
+    }
 
-    fn import_block(&mut self, header: &BlockHeader) {
+    fn import_block_header(&mut self, header: &BlockHeader) {
         self.headers.insert(header.number, header.clone());
     }
 
+    fn import_block_body(&mut self, body: &BlockBody) {
+        info!("Received block body, ignoring.");
+    }
+
```

Comment:
Note that reth-scheduler needs to be the one that ties block header/body pair and only after it is paired and in right order send to client for verification. I see this as part of `ActiveSync`.

Reply:
Okay, so I will change the `Blockchain` trait back to having just an `import_block` in the first PR on syncing.